test/ln_test.py

```diff
@@ -7,99 +7,118 @@
 from test_base import TestBase
 from warnet.services import ServiceType
 
-graph_file_path = Path(os.path.dirname(__file__)) / "data" / "ln.graphml"
 
-base = TestBase()
-base.start_server()
-
-
-def get_cb_forwards(index):
-    cmd = "wget -q -O - 127.0.0.1:9235/api/forwarding_history"
-    res = base.wait_for_rpc(
-        "exec_run", [index, ServiceType.CIRCUITBREAKER.value, cmd, base.network_name]
-    )
-    return json.loads(res)
-
-
-print(base.warcli(f"network start {graph_file_path}"))
-base.wait_for_all_tanks_status(target="running")
-base.wait_for_all_edges()
-
-print("\nRunning LN Init scenario")
-base.warcli("rpc 0 getblockcount")
-base.warcli("scenarios run ln_init")
-base.wait_for_all_scenarios()
-
-node2pub, node2host = json.loads(base.warcli("lncli 2 getinfo"))["uris"][0].split("@")
-
-print("\nEnsuring node-level channel policy settings")
-chan_id = json.loads(base.warcli("lncli 2 listchannels"))["channels"][0]["chan_id"]
-chan = json.loads(base.warcli(f"lncli 2 getchaninfo {chan_id}"))
-# node_1 or node_2 is tank 2 with its non-default --bitcoin.timelockdelta=33
-if chan["node1_policy"]["time_lock_delta"] != 33:
-    assert chan["node2_policy"]["time_lock_delta"] == 33
-
-print("\nEnsuring no circuit breaker forwards yet")
-assert len(get_cb_forwards(1)["forwards"]) == 0
-
-print("\nTest LN payment from 0 -> 2")
-inv = json.loads(base.warcli("lncli 2 addinvoice --amt=2000"))["payment_request"]
-
-print(f"\nGot invoice from node 2: {inv}")
-print("\nPaying invoice from node 0...")
-print(base.warcli(f"lncli 0 payinvoice -f {inv}"))
-
-print("Waiting for payment success")
-
-
-def check_invoices():
-    invs = json.loads(base.warcli("lncli 2 listinvoices"))["invoices"]
-    if len(invs) > 0 and invs[0]["state"] == "SETTLED":
-        print("\nSettled!")
-        return True
-    else:
+class LNTest(TestBase):
+    def __init__(self):
+        super().__init__()
+        self.graph_file_path = Path(os.path.dirname(__file__)) / "data" / "ln.graphml"
+
+    def run_test(self):
+        self.start_server()
+        try:
+            self.setup_network()
+            self.run_ln_init_scenario()
+            self.test_channel_policies()
+            self.test_ln_payment_0_to_2()
+            self.test_ln_payment_2_to_0()
+            self.test_simln()
+        finally:
+            self.stop_server()
+
+    def setup_network(self):
+        self.log.info("Setting up network")
+        print(self.warcli(f"network start {self.graph_file_path}"))
```

Comment:
cdc1de058b6efe42b55b30ae63e25b6ad1ffa434

damn i haven't been checking them all but i think this should be a log message not print.

Reply:
fixed all, thanks